### PR TITLE
Add support of lambdas as CLOSURE scope type

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do

# Motivation

# Additional Notes

Jira ticket: [DEBUG-1892]

[DEBUG-1892]: https://datadoghq.atlassian.net/browse/DEBUG-1892?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/symbol/SymbolExtractionTransformerTest.java [hunk 1]
```diff
@@ -446,42 +446,84 @@ public void symbolExtraction09() throws IOException, URISyntaxException {
...
+    assertScope(supplierClosureLocalScope, ScopeType.LOCAL, null, 20, 21, SOURCE_FILE, 0, 1);
+    assertSymbol(
+        supplierClosureLocalScope.getSymbols().get(0),
+        SymbolType.LOCAL,
+        "var1",
+        Integer.TYPE.getTypeName(),
+        20);
+    Scope lambdaClosureScope = classScope.getScopes().get(4);
+    assertScope(lambdaClosureScope, ScopeType.CLOSURE, "lambda$main$0", 11, 12, SOURCE_FILE, 1, 1);
+    assertSymbol(
```
Comment (ojung) [inline dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/symbol/SymbolExtractionTransformerTest.java:511]: Can you also assert that it doesn't have `outside2` scope?

**Review by ojung — APPROVED** [review]
Awesome, thanks 👍

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/symbol/SymbolExtractionTransformerTest.java (hunk 1 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/symbol/SymbolExtractionTransformerTest.java:511]: indirectly this is the case. we have only 1 symbol in this scope and this symbol is `outside`

Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/symbol/SymbolExtractionTransformerTest.java:511]: on `assertScope` the last number is the expected number of symbols inside this scope

**Pull request merged**